go-controller/pkg/node/node.go

```diff
@@ -201,40 +208,43 @@ func (n *OvnNode) Start(wg *sync.WaitGroup) error {
 		return fmt.Errorf("timed out waiting for node's: %q logical switch: %v", n.name, err)
 	}
 
-	klog.Infof("Node %s ready for ovn initialization with subnet %s", n.name, util.JoinIPNets(subnets, ","))
+	if config.OvnKubeNode.Mode != types.NodeModeSmartNICHost {
+		klog.Infof("Node %s ready for ovn initialization with subnet %s", n.name, util.JoinIPNets(subnets, ","))
 
-	if _, err = isOVNControllerReady(n.name); err != nil {
-		return err
-	}
+		if _, err = isOVNControllerReady(n.name); err != nil {
+			return err
+		}
 
-	nodeAnnotator := kube.NewNodeAnnotator(n.Kube, node)
-	waiter := newStartupWaiter()
+		nodeAnnotator := kube.NewNodeAnnotator(n.Kube, node)
+		waiter := newStartupWaiter()
 
-	// Initialize management port resources on the node
-	mgmtPortConfig, err := createManagementPort(n.name, subnets, nodeAnnotator, waiter)
-	if err != nil {
-		return err
-	}
+		// Initialize management port resources on the node
+		mgmtPortConfig, err = createManagementPort(n.name, subnets, nodeAnnotator, waiter)
+		if err != nil {
+			return err
+		}
 
-	// Initialize gateway resources on the node
-	if err := n.initGateway(subnets, nodeAnnotator, waiter, mgmtPortConfig); err != nil {
-		return err
-	}
+		// Initialize gateway resources on the node
+		if err := n.initGateway(subnets, nodeAnnotator, waiter, mgmtPortConfig); err != nil {
+			return err
+		}
 
-	if err := nodeAnnotator.Run(); err != nil {
-		return fmt.Errorf("failed to set node %s annotations: %v", n.name, err)
-	}
+		if err := nodeAnnotator.Run(); err != nil {
+			return fmt.Errorf("failed to set node %s annotations: %v", n.name, err)
+		}
+		go n.gateway.Run(n.stopChan, wg)
 
-	// Wait for management port and gateway resources to be created by the master
-	klog.Infof("Waiting for gateway and management port readiness...")
-	start := time.Now()
-	if err := waiter.Wait(); err != nil {
-		return err
+		// Wait for management port and gateway resources to be created by the master
+		klog.Infof("Waiting for gateway and management port readiness...")
+		start := time.Now()
+		if err := waiter.Wait(); err != nil {
+			return err
+		}
+		klog.Infof("Gateway and management port readiness took %v", time.Since(start))
 	}
-	go n.gateway.Run(n.stopChan, wg)
-	klog.Infof("Gateway and management port readiness took %v", time.Since(start))
 
 	if config.HybridOverlay.Enabled {
+		// Not supported with Smart-NIC, enforced in config
```

Comment:
should we add TODO and revisit this in the feature?

Reply:
TODO added